packages/flutter/lib/src/material/data_table.dart

```diff
@@ -851,15 +895,47 @@ class DataTable extends StatelessWidget {
       ?? dataTableTheme.dataRowHeight
       ?? themeData.dataTableTheme.dataRowHeight
       ?? kMinInteractiveDimension;
-    label = Container(
-      padding: padding,
-      height: effectiveDataRowHeight,
-      alignment: numeric ? Alignment.centerRight : AlignmentDirectional.centerStart,
-      child: DefaultTextStyle(
-        style: effectiveDataTextStyle.copyWith(
-          color: placeholder ? effectiveDataTextStyle.color!.withOpacity(0.6) : null,
+
+    String semanticsLabel = '';
+    if (rowHeading != null) {
+      semanticsLabel += '$rowHeading, ';
+    }
+    if (columnHeading != null) {
+      semanticsLabel += '$columnHeading, ';
+    }
+    final MaterialLocalizations localizations = MaterialLocalizations.of(context);
+    final String tableCoordinatesLabel = localizations.tableCoordinatesLabel('$rowNumber', '$totalRows', '$columnNumber', '$totalColumns');
+    // Each cell should have it own SemanticsNode, which I found is not always
+    // the case through data_table_test.dart. Thus added the outer Semantics
+    // widget with container = true.
+    label = Semantics(
+      container: true,
+      child: Container(
+        padding: padding,
+        height: effectiveDataRowHeight,
+        alignment: numeric ? Alignment.centerRight : AlignmentDirectional.centerStart,
+        child: Stack(
+          children: <Widget>[
+            // This Semantics widget is for announcing [rowHeading] and [columnHeading]
+            // (wrapped in [semanticsLabel]) before the actual content of the [DataCell],
+            // which is wrapped in [DefaultTextStyle]
+            Semantics(
+              label: semanticsLabel,
+              onTapHint: localizations.tableCellOnTapHint,
```

Comment:
It is weird that we set the onTaphint here without the ontap action, and I can't find where the ontap to select the row is define, can you point me to the code?